kickstarter/android-oss
# MBL-1587: Unify add ons selection screen for crowdfund and late pledges user journeys.

# 📲 What

A description of the change.

# 🤔 Why

Some background context on why the change is needed.

# 🛠 How

More in-depth discussion of the change or implementation.

# 👀 See

- Crowdfund checkout using compose `AddOnsScreen` and `AddOnsViewModel` with new states architecture + coroutines, for pledge and update pledge flows

https://github.com/user-attachments/assets/40b5da83-3e5e-4832-910b-b6176ddb19c1

- Late pledges using compose `AddOnsScreen` and `AddOnsViewModel` with all modifications required to be able to unify with crowdfund


https://github.com/user-attachments/assets/225fafdd-2fd5-4589-a5d2-b2bb653a422b


| --- | --- |
|  |  |

# 📋 QA
- Pledge to any project in staging selecting addons, make sure your addon selection gets carried over to the checkout screen
- Update some of your pledged project rewards modifying your addOns selection, make sure your addOns selection gets carried over to the checkout screen
- Go to a late pledge enabled project (or modify in ProjectExt:showLatePledgeFlow to always be true so every project will load late pledges flow) with available addOns, make sure your addOns selection gets carried over to the checkout screen 
- Make sure you can see the `pageviewd` analytic events in segment for both crowdfund and late pledges

# Story 📖

[Name of Trello Story](Trello link)
